File: readability.py
from cs50 import get_string

# Get the input to evaluate
text = get_string("Text: ")
# Calculate words based on spaces in the input string
words = len(text.split())

# Calculate sentences based on terminators
full_stops = len(text.split(". "))
questions = len(text.split("? "))

# In case of question marks, there will be one full stop sentence extra to be removed from the sum
if (questions > 1):
    sentences = full_stops + questions - 1
else:
    sentences = full_stops

# Calculate number of letters
letters = 0
for i in text:
    if (i.isalpha()):
        letters += 1

# Calculate average letters and sentences by words
avg_letters = 100 * (letters/words)
avg_sentences = 100 * (sentences/words)

# Evaluate grade using Coleman-Liau index
grade = round(0.0588 * avg_letters - 0.296 * avg_sentences - 15.8)

# Print the grade
if (grade > 16):
    print("Grade 16+")
elif (grade < 1):
    print("Before Grade 1")
else:
    print(f"Grade {grade}")

File: readability.c
#include <ctype.h>
#include <stdio.h>
#include <string.h>

// Function to count the number of letters in the text
int count_letters(char text[])
{
    int letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

// Function to count the number of words in the text
int count_words(char text[])
{
    int words = 1; // Start with 1 to account for the first word
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }
    return words;
}

// Function to count the number of sentences in the text
int count_sentences(char text[])
{
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }
    return sentences;
}

int main(void)
{
    char text[1000];

    // Prompt the user to enter text
    printf("Text: ");
    fgets(text, sizeof(text), stdin); // Using fgets for multi-word input with spaces

    // Calculate the number of letters, words, and sentences
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Calculate L and S
    float L = (float) letters / words * 100;
    float S = (float) sentences / words * 100;

    // Calculate the Coleman-Liau index
    float index = 0.0588 * L - 0.296 * S - 15.8;

    // Output the result
    int grade = (int) (index + 0.5); // Round the index to the nearest integer

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", grade);
    }

    return 0;
}
